frontend/public/components/nav/admin-nav.tsx

```diff
@@ -85,193 +87,196 @@ const MonitoringNavSection_ = ({ canAccess }) => {
+      <NavSection title="Pipelines" />
 
-    <NavSection title="Service Catalog" required={FLAGS.SERVICE_CATALOG}>
-      <HrefLink
-        href="/provisionedservices"
-        name="Provisioned Services"
-        activePath="/provisionedservices/"
-        startsWith={provisionedServicesStartsWith}
-      />
-      <HrefLink
-        href="/brokermanagement"
-        name="Broker Management"
-        activePath="/brokermanagement/"
-        startsWith={brokerManagementStartsWith}
-      />
-    </NavSection>
+      <NavSection title="Service Catalog" required={FLAGS.SERVICE_CATALOG}>
+        <HrefLink
+          href="/provisionedservices"
+          name="Provisioned Services"
+          activePath="/provisionedservices/"
+          startsWith={provisionedServicesStartsWith}
+        />
+        <HrefLink
+          href="/brokermanagement"
+          name="Broker Management"
+          activePath="/brokermanagement/"
+          startsWith={brokerManagementStartsWith}
+        />
+      </NavSection>
 
-    <MonitoringNavSection />
+      <MonitoringNavSection />
 
-    <NavSection title="Compute" required={FLAGS.CAN_LIST_NODE}>
-      <ResourceClusterLink resource="nodes" name="Nodes" />
-      <HrefLink
-        href={formatNamespacedRouteForResource(
-          referenceForModel(MachineModel),
-          'openshift-machine-api',
-        )}
-        name="Machines"
-        required={FLAGS.CLUSTER_API}
-      />
-      <HrefLink
-        href={formatNamespacedRouteForResource(
-          referenceForModel(MachineSetModel),
-          'openshift-machine-api',
-        )}
-        name="Machine Sets"
-        required={FLAGS.CLUSTER_API}
-      />
-      <HrefLink
-        href={formatNamespacedRouteForResource(
-          referenceForModel(MachineAutoscalerModel),
-          'openshift-machine-api',
-        )}
-        name="Machine Autoscalers"
-        required={FLAGS.MACHINE_AUTOSCALER}
-      />
-      <HrefLink
-        href={formatNamespacedRouteForResource(
-          referenceForModel(MachineHealthCheckModel),
-          'openshift-machine-api',
-        )}
-        name="Machine Health Checks"
-        required={FLAGS.MACHINE_HEALTH_CHECK}
-      />
-      <Separator required={FLAGS.MACHINE_CONFIG} name="ComputeSeparator" />
-      <ResourceClusterLink
-        resource={referenceForModel(MachineConfigModel)}
-        name="Machine Configs"
-        required={FLAGS.MACHINE_CONFIG}
-      />
-      <ResourceClusterLink
-        resource={referenceForModel(MachineConfigPoolModel)}
-        name="Machine Config Pools"
-        required={FLAGS.MACHINE_CONFIG}
-      />
-    </NavSection>
+      <NavSection title="Compute" required={FLAGS.CAN_LIST_NODE}>
+        <ResourceClusterLink resource="nodes" name="Nodes" />
+        <HrefLink
+          href={formatNamespacedRouteForResource(
+            referenceForModel(MachineModel),
+            lastNamespace === ALL_NAMESPACES_KEY ? lastNamespace : 'openshift-machine-api',
```

Comment:
Don't duplicate this logic for each nav item. Instead have something like

```ts
const machineNS = lastNamespace === ALL_NAMESPACES_KEY ? lastNamespace : 'openshift-machine-api';
```

And then use `machineNS` for the four nav items.

Worth adding a comment explaining the reason as well.

Reply:
All set.